docs_src/design/adr/014-Secret-Provider-For-All.md

```diff
@@ -0,0 +1,409 @@
+
+Application Services currently have the ability to configure `SecretStores` for **Service Exclusive** and/or **Service Shared** secrets depending on their needs.
+
+#### Known and Unknown Services 
+
+- **Known Services** are those identified in the static configuration by security-secretstore-setup
+  
+  - These currently are Core Data, Core Metadata, Support Notifications, Support Scheduler and Application Service (class)
+  
+- **Unknown Services** are those not known in the static configuration that become known when added to the Docker compose file or Snap. 
+
+  - Application Service (instance) are examples of these services. 
+
+  - Service exclusive `SecretStore` can be created for these services by adding the services' unique name , i.e. appservice-http-export, to the `ADD_SECRETSTORE_TOKENS` environment variable for security-secretstore-setup
+
+    ```
+    ADD_SECRETSTORE_TOKENS: "appservice-http-export, appservice-mqtt-export"
+    ```
+
+    This creates an exclusive `SecretStore` for each service listed. The name provided for each service must be used in the service's `SecretStore` configuration and Docker volume mount. Typically the configuration is set via environment overrides or is already in an existing configuration profile (***http-export*** profile for app-service-configurable). 
+
+    Example docker-compose file entries:
+
+    ```yaml
+    environment:
+        ...
+    	SecretStoreExclusive_Path: "/v1/secret/edgex/appservice-http-export/"
+    	TokenFile: "/tmp/edgex/secrets/appservice-http-export/secrets-token.json"
+    	
+    volumes:
+    	...
+    	- /tmp/edgex/secrets/appservice-http-export:/tmp/edgex/secrets/appservice-http-export:ro,z
+    ```
+
+#### Known and Unknown Secrets
+
+- **Known Secrets** are those identified by name in the static configuration whose values are randomly generated at seed time. These secrets are seeded on start-up of EdgeX.
+  - Database credentials are currently the only secrets of this type
+
+- **Unknown Secrets** are those not known in the static configuration and that become known during run time. These secrets are seeded at run time via the Application Services `/api/vX/Secrets` endpoint
```

Comment:
Known and Unknown names here seem confusing to me.  What about SeedTimeSecrets and RuntimeSecrets?  Both are randomly generated - it is just a matter of when.

Reply:
Nope, they are not both randomly generated, Unknow Secrets are things like and Auth token for REST call to export data to an endpoint. Both are seeded, static Known ones at startup and Unknown ones when ever the user pushes (seeds) the secret via the `Secrets` endpoint.

How about `Static Defined Secrets` and `Runtime Secrets`?

Reply:
Changed to  `Static Secrets and Runtime Secrets`